spec/requests/api/guest_devices_spec.rb

```diff
@@ -0,0 +1,32 @@
+RSpec.describe "guest devices API" do
+  describe "display guest device details" do
+    context "with a valid role" do
+      it "shows its properties" do
+        device = FactoryGirl.create(:guest_device,
+                                    :device_name => "Broadcom 2-port 1GbE NIC Card",
+                                    :device_type => "ethernet",
+                                    :location    => "Bay 7")
+
+        api_basic_authorize action_identifier(:guest_devices, :read, :resource_actions, :get)
+
+        run_get(guest_devices_url(device.id))
+
+        expect_single_resource_query("device_name" => "Broadcom 2-port 1GbE NIC Card",
+                                     "device_type" => "ethernet",
+                                     "location"    => "Bay 7")
+      end
+    end
+
+    context "with an invalid role" do
```

Comment:
Nit, but wording here is misleading. It's not that the role is invalid, it's that the user's role does not authorize them to use this feature. Similarly, it doesn't fail - it does the right thing! i.e. respond with a forbidden status. Let's merge this as-is though, maybe you could consider rewording this in a follow up?